release/models/platform/openconfig-platform-storage.yang

```diff
@@ -0,0 +1,75 @@
+module openconfig-platform-storage {
+
+  yang-version "1";
+
+  // namespace
+  namespace "http://openconfig.net/yang/platform/storage";
+
+  prefix "oc-storage";
+
+  import openconfig-platform {
+    prefix oc-platform;
+  }
+  import openconfig-extensions {
+    prefix oc-ext;
+  }
+  import openconfig-yang-types {
+    prefix oc-yang;
+  }
+
+  // meta
+  organization
+    "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+     www.openconfig.net";
+
+  description
+    "This module defines data related to STORAGE components in the
+     OpenConfig platform model.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2024-04-24" {
+    description
+      "Initial revision.";
+    reference
+      "0.1.0";
+  }
+
+  // OpenConfig specific extensions for module metadata.
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  // identity statements
+  // typedef statements
+  // grouping statements
+  grouping storage-state {
+    description
+      "Storage component state.  These counters are derived from the
+       linux kernel block layer statistics in /sys/block/<dev>/stat.";
```

Comment:
Thx for updating this.  A few comments:

* Confirming that since this is the first use of `/components/component/storage` that it is now expected that each storage device (at least block dev) such as those enumerated w/ `lsblk` be represented as a component of type `STORAGE` w/ a respective parent of the physical component housing the storage device (The relationship of containers to respective types is not drawn out in the modeling but rather an implication of when such containers should be in use depending on the type of component)
* This and various other models will often assume and imply an underlying Linux OS which while in most cases is true, is not guaranteed but it would be best to then blanket such implementation assumptions with Linux being the mandatory underlying OS

Reply:
We'll introduce a [featureprofiles test ](https://github.com/openconfig/featureprofiles) which will reference this path and may provide additional expectations on related leaves, such as `parent`.  

We don't always have detailed expectations for all possible related leaves.  At the moment, we don't have an operational use case to model the component tree for storage devices, so I expect this to remain undefined.

Regarding the linux reference, I'll loosen the description slightly to state that linux is a reference example and should be used if applicable.  Other implementations count I/O operations which cannot be completed due to an issue with the storage device.